### PR TITLE
At 521 enable authentication for opensearch

### PR DESCRIPTION
**Pull request by larox11**
### Added:
  - added support for openID authentication type. user can now switch between basic auth and openID connect.
  - added new configuration variables
    ```
	AuthUsername string `validate:"required" viperEnv:"ELASTIC_AUTH_USER"`
	AuthPassword string `validate:"required" viperEnv:"ELASTIC_AUTH_PASS"`
	AuthMethod   string `validate:"required" viperEnv:"ELASTIC_AUTH_METHOD"`
    ```

### Changed:
  - upgraded github.com/opensearch-project/opensearch-go to v2 v2.3.0

## References

https://jira.greenbone.net/browse/AT-521

## Checklist

- [x] Tests
- [x] Documentation

### CODE REVIEW

File: pkg/openSearch/openSearchClient/updateQueue.go [hunk 1]
```diff
@@ -31,26 +30,26 @@ type Request struct {
 
 // UpdateQueue is a queue for OpenSearch update requests.
 type UpdateQueue struct {
-	openSearchProjectClient *opensearch.Client
-	queue                   chan *Request
-	stop                    chan bool
-	wg                      sync.WaitGroup
-	updateMaxRetries        int
-	updateRetryDelay        time.Duration
+	client           esapi.Transport
```
[the remark this answers is not in the document]
Comment (larox11) [reply pkg/openSearch/openSearchClient/updateQueue.go]: good idea! esapi could easily be replaced with opensearchapi: https://github.com/greenbone/opensight-golang-libraries/pull/72/commits/df94a8fdd4da7c1ca35b3d4cd9d8db390a4dad49

File: pkg/openSearch/openSearchClient/authenticator.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	case basic:
+		reqClone.SetBasicAuth(a.config.AuthUsername, a.config.AuthPassword)
+	case openId:
+		token, err := a.tokenReceiver.GetClientAccessToken(a.config.AuthUsername, a.config.AuthPassword)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve authorization header: %w", err)
+		}
+		reqClone.Header.Set("Authorization", "Bearer "+token)
+	default:
+		log.Error().Msgf("undefined authentication method for opensearch client: %s", a.authMethod)
```
[the remark this answers is not in the document]
Comment (larox11) [reply pkg/openSearch/openSearchClient/authenticator.go]: done: https://github.com/greenbone/opensight-golang-libraries/pull/72/commits/29834b28b32f42646037784dcaf6ada59cc7977d

File: pkg/openSearch/openSearchClient/authenticator_test.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+			authMethod:     openId,
+			authUser:       "clientID",
+			authPass:       "clientSecret",
+			mockToken:      "mockToken",
+			mockError:      nil,
+			expectedHeader: "Bearer mockToken",
+		},
+		{
+			name:           "returns error when tokenReceiver.GetClientAccessToken fails",
+			authMethod:     openId,
```
[the remark this answers is not in the document]
Comment (larox11) [reply pkg/openSearch/openSearchClient/authenticator_test.go:144]: done: https://github.com/greenbone/opensight-golang-libraries/pull/72/commits/17fb704280fa1de6b9e0e3705a72e1e4a5f3442b

File: pkg/openSearch/openSearchClient/authenticator_test.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+				AuthMethod:   "openid",
+				AuthUsername: "clientID",
+				AuthPassword: "clientSecret",
+			},
+			tokenReceiver:  &MockTokenReceiver{},
+			expectedMethod: openId,
+			expectedError:  false,
+		},
+		{
+			name: "returns error when password is empty",
```
[the remark this answers is not in the document]
Comment (larox11) [reply pkg/openSearch/openSearchClient/authenticator_test.go:69]: done: https://github.com/greenbone/opensight-golang-libraries/pull/72/commits/1fc1165af6e88e16d569d2a6bd8aa9bd8b0bf6de

**Review by mgoetzegb — APPROVED** [review]

**Review by HollererJ — APPROVED** [review]

**Pull request merged**